world.js

```diff
@@ -283,13 +288,33 @@ class World {
   // ----------------- Spawn Functions -----------------
   
 
-  /** Spawn a number of nematodes
-   * 
-   * @param {*} number The number of nematodes to spawn
+  /* Spawn a number of nematodes
+   * number: the number of nematodes to spawn
    */
   SpawnNematodes(number){
     for (let i = 0; i < number; i++) {
-        world.selectedNematode = new Nematode()
+      world.selectedNematode = new Nematode()
+    }
+  }
+
+  /* Spawn a number of "Smart" nematodes
+   * number: the number of nematodes to spawn
+   */
+  SpawnSmartNematodes(number){
+
+    for (let i = 0; i < number; i++) {
+
+      // Load a random nematode from the training data
+      const nematodeData = NematodeTrainer.GetRandomNematode();
+
+      // Create a new nematode with the loaded data
+      const parent = new Nematode(nematodeData);
```

Comment:
why are we creating two nematodes here?

Reply:
I create the parent, then create a new child and destroy the parent. I wanted smart nematodes to not be direct copies of previous nematodes. Slightly inefficient, but won't be an issue. I could make a "CreateChildFromJson" function, but that would be more duplicated code that is not really needed.